src/Data/Vector/Hashtables/Internal.hs

```diff
@@ -70,9 +70,9 @@ data Dictionary_ s ks k vs v = Dictionary {
     hashCode,
     next,
     buckets,
-    refs :: IntArray s,
-    key :: ks s k,
-    value :: vs s v
+    refs :: !(IntArray s),
```

Comment:
How's about we just do StrictData in this module? All these bangs are a bit obscuring imo.

Reply:
I prefer explicit bangs, because otherwise it's too easy to move code during refactoring to another module without `StrictData` and silently break performance. But I'm happy to comply with your wishes.

Reply:
Good point. We can leave it as is.